pkg/trace/config/config.go

```diff
@@ -197,14 +197,14 @@ func (c *AgentConfig) APIKey() string {
 }
 
 // Validate validates if the current configuration is good for the agent to start with.
-func (c *AgentConfig) validate() error {
+func (c *AgentConfig) validate(hostNameResolution bool) error {
```

Comment:
Please don't take this approach. Simply just add a new boolean field to `*AgentConfig` called `DisableGRPCHostname` and return from `acquireHostname` early with an error if it's set:
```go
return errors.new("disabled by config")
```

Reply:
Oh, wait. You can't do that.